x-pack/metricbeat/module/gcp/metrics/metricset.go

```diff
@@ -352,3 +359,35 @@ func addHostFields(groupedEvents []KeyValuePoint) mapstr.M {
 	}
 	return hostRootFields
 }
+
+func (m *MetricSet) setOrganization() error {
+	// get project details
+	ctx := context.Background()
+	// Initialize the Cloud Resource Manager service
+	srv, err := cloudresourcemanager.NewService(ctx, m.config.opt...)
+	if err != nil {
+		return fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
+	}
+
+	// Get the project ancestor details
+	ancestryResponse, err := srv.Projects.GetAncestry(m.config.ProjectID, &cloudresourcemanager.GetAncestryRequest{}).Do()
+	if err != nil {
+		return fmt.Errorf("failed to get project ancestors: %w", err)
+	}
+
+	ancestor := ancestryResponse.Ancestor[len(ancestryResponse.Ancestor)-1]
+	if ancestor.ResourceId.Type == "organization" {
+		m.config.OrganizationID = ancestor.ResourceId.Id
+		orgReq := srv.Organizations.Get(fmt.Sprintf("organizations/%s", m.config.OrganizationID))
+
+		orgDetails, err := orgReq.Do()
+		if err != nil {
+			return fmt.Errorf("failed to get organization details: %w", err)
+		}
+		if orgDetails.DisplayName != "" {
+			m.config.OrganizationName = orgDetails.DisplayName
+		}
+	}
+
+	return nil
+}
```

Comment:
Done

Reply:
@Linu-Elias Please review the refactored function. In the first version, the check `if orgDetails.DisplayName != ""` is redundant since it's already handled when adding the ECS labels. Also checking the length of `ancestryResp.Ancestor` should be done to avoid panics if there are no ancestors.

Reply:
Done